pkg/virtctl/create/preference/preference_test.go

```diff
@@ -140,20 +146,44 @@ func setFlag(flag, parameter string) string {
 	return fmt.Sprintf("--%s=%s", flag, parameter)
 }
 
-func getPreferenceSpec(bytes []byte, namespaced bool) (*instancetypev1beta1.VirtualMachinePreferenceSpec, error) {
+func testAdmissionAndGetSpec(bytes []byte, namespaced bool) (*instancetypev1beta1.VirtualMachinePreferenceSpec, error) {
```

Comment:
a function should do one thing. The fact that you are force to add `And` to its name is a code smell.

How about returning in error instead of `Expect`ing ?

Reply:
Right, these assertions should not be part of the same helper function.